crates/miden-lib/src/transaction/memory.rs

```diff
@@ -124,20 +124,29 @@ pub const ACCT_ID_PTR: MemoryAddress = 404;
 /// The memory address at which the initial account commitment is stored.
 pub const INIT_ACCT_COMMITMENT_PTR: MemoryAddress = 408;
 
-/// The memory address at which the input notes commitment is stored.
-pub const INPUT_NOTES_COMMITMENT_PTR: MemoryAddress = 412;
-
 /// The memory address at which the initial nonce is stored.
-pub const INIT_NONCE_PTR: MemoryAddress = 416;
+pub const INIT_NONCE_PTR: MemoryAddress = 412;
+
+/// The memory address at which the initial account vault root is stored.
+pub const INIT_ACCT_VAULT_ROOT_PTR: MemoryAddress = 416;
+
+/// The memory address at which the initial account storage commitment is stored.
+pub const INIT_ACCT_STORAGE_COMMITMENT_PTR: MemoryAddress = 420;
+
+/// The memory address at which the initial account code commitment is stored.
+pub const INIT_ACCT_CODE_COMMITMENT_PTR: MemoryAddress = 424;
```

Comment:
I think we handle account code a bit differently fro the rest. Specifically, we never update the code in the actual account data - so, effectively, account data always has the initial code commitment.

We don't really support this yet, but the new code commitment is stored separately, in the memory address pointed to by `NEW_CODE_ROOT_PTR` defined on line 92 above.

This is because we don't want to change the code during transaction execution, and only update it after the transaction is done executing (though, I believe, this mechanism doesn't fully work right now).

Reply:
I think we can get rid of this constant, right?
We should be able to use `memory::get_acct_code_commitment` instead of `INIT_ACCT_CODE_COMMITMENT_PTR`.

The code related to the "new code commitment" can stay as-is, but we don't use it as part of this PR afaict.

Reply:
I would leave this constant just for the convenience sake, we will have all out initial native account commitments in one place, but probably it makes sense just to make this constant point to the code commitment of the native account, as Bobbin suggested in one of the comments below: 
`INIT_ACCT_CODE_COMMITMENT_PTR = NATIVE_ACCOUNT_DATA_PTR + ACCT_CORE_DATA_SECTION_END_OFFSET`